platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -146,6 +184,215 @@ null_crypto_routine(odp_packet_t pkt ODP_UNUSED,
 	return ODP_CRYPTO_ALG_ERR_NONE;
 }
 
+static
+odp_crypto_alg_err_t aes_gcm_encrypt(odp_packet_t pkt,
+				     const odp_crypto_packet_op_param_t *param,
+				     odp_crypto_generic_session_t *session)
+{
+	armv8_cipher_state_t cs = {
+		.counter = {
+			.d = {0, 0}
+		}
+	};
+	uint8_t *iv_ptr;
+	uint64_t iv_bit_length;
+	uint64_t plaintext_bit_length = param->cipher_range.length * 8;
+	uint64_t aad_bit_length = session->p.auth_aad_len * 8;
+	uint32_t in_pos = param->cipher_range.offset;
+	uint32_t in_len = param->cipher_range.length;
+	int ret = 0;
+
+	/* Fail early if cipher_range is too large */
+	if (in_len > ARM_CRYPTO_MAX_DATA_LENGTH) {
+		ODP_DBG("ARM Crypto: Packet size too large for requested operation\n");
+		ret = -1;
+		goto err;
+	}
+
+#if ODP_DEPRECATED_API
+	if (param->cipher_iv_ptr)
+		iv_ptr = param->cipher_iv_ptr;
+	else if (session->p.cipher_iv.data)
+		iv_ptr = session->cipher.iv_data;
+	else
+		return ODP_CRYPTO_ALG_ERR_IV_INVALID;
+
+	iv_bit_length = session->p.cipher_iv.length * 8;
+#else
+	iv_ptr = param->cipher_iv_ptr;
+	iv_bit_length = session->p.cipher_iv_len * 8;
```

Comment:
You could just have the iv_bit_length initialized at the variable declaration like you did in the previous version. There is no need to use cipher_iv.length anymore.

Having deprecated API enabled at ODP compilation does not mean that applications would only use the deprecated APIs when they have a choise. They may use a deprecated API for some other feature but not this one or there may be an application that does not use any of the deprecated APIs. Thus ODP implementation should always read IV length from the cipher_iv_len parameter, even when compiled with deprecated APIs enabled. The API has been defined so that cipher_iv_len and cipher_iv.length are in the same memory location so things work even if an application is using the deprecated API. So, do not use cipher_iv.length or auth_iv_length anymore in this file.

Reply:
I will remove this usage from both the functions and initialize it like I did before. Thanks!